src/init.c

```diff
@@ -255,7 +260,7 @@ static bool _mi_heap_done() {
 // to set up the thread local keys.
 // --------------------------------------------------------
 
-#ifndef _WIN32
+#if !defined(_WIN32) && !defined(__wasi__)
```

Comment:
Or maybe

```diff
 
-#ifndef _WIN32
+#ifdef __wasi__
+/* no pthreads in WebAssembly */
+#elif !defined(_WIN32)
 #define MI_USE_PTHREADS
```

That way, it will be much harder to break this in future PRs.

Reply:
Done.